collect_app/src/main/java/org/odk/collect/android/formentry/FormEndViewFactory.kt

```diff
@@ -0,0 +1,33 @@
+package org.odk.collect.android.formentry
+
+import android.content.Context
+import org.odk.collect.android.instancemanagement.autosend.AutoSendSettingsProvider
+import org.odk.collect.forms.Form
+import org.odk.collect.forms.shouldFormBeSentAutomatically
+import org.odk.collect.settings.SettingsProvider
+import org.odk.collect.settings.keys.ProtectedProjectKeys
+
+class FormEndViewFactory(
```

Comment:
I don't dislike this structure at all, but it does feel like the kind of logic we'd more typically move into a `ViewModel`. I could imagine adding something like this to the current `FormEntryViewModel` interface:

```kotlin
fun isSaveDraftEnabled(): Boolean
fun isFinalizeEnabled(): Boolean
fun shouldFormBeSentAutomatically(): Boolean
```

The `FormEndView` could then simply take the `FormEntryViewModel` as a constructor argument and the tests in `FormEndViewFactory` could just be tests for `FormEndView` itself (with a stub `FormEntryViewModel` being used).

Reply:
Yeah, I've implemented it like this because I think about that `End view` as a separate unit that could even be a separate fragment in the future. For example, we could have `FormFillingFragment`, `FormHierarchyFragment` and `FormEndFragment`. Maybe that's why I didn't want to mix that logic and add it to one of the existing view models... hmm maybe a good solution, for now, would be creating a separate view model for that end view which would hold those methods and in the future we could easily move it from `FormFillingActivity` to `FormEndFragment`?

Reply:
Yeah, happy for it to be a separate `ViewModel` if it doesn't feel connected at the moment. `FormEntryViewModel` is already pretty big, and I don't think this shares any state, so I do think you're right to try and keep them apart.

Reply:
Done.